### PR TITLE
Issue #9341: Fix Try child suggesting wrong indentation level

### LINKED ISSUE #9341: Checkstyle 8.41 introduces regression on indentation rules when the line starts with new after try {
I have read check documentation: https://checkstyle.org/config_misc.html#Indentation
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac CheckstyleBug.java
/var/tmp $ cat CheckstyleBug.java
public class CheckstyleBug {
    public void delete(final int assetId) {
        try {
            new Object().equals(Integer.valueOf(assetId));
        } catch (Exception ignore) {
            // Ignore, but log failure to delete views.
        }
    }
}


/var/tmp $ cat checkstyle.xml
<?xml version="1.0"?>
<module name="Checker">
       .....
        <module name="Indentation">
            <property name="basicOffset" value="4"/>
            <property name="braceAdjustment" value="0"/>
            <property name="caseIndent" value="4"/>
            <property name="throwsIndent" value="8"/>
            <property name="lineWrappingIndentation" value="8"/>
            <property name="arrayInitIndent" value="2"/>
            <property name="forceStrictCondition" value="true"/>
        </module>
....
</module>

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-8.41-all.jar -c checkstyle.xml CheckstyleBug.java 
Starting audit...
[WARN] /Users/john.camerin/Projects/accelerator-merge-master/CheckstyleBug.java:4:13: 'new' has incorrect indentation level 12, expected level should be 8. [Indentation]
Audit done.

```

The rule is expecting the 'new' keyword to have the same indentation as the 'try' block within which it is contained.  The try keyword is followed by curly braces, which should apply the basicOffset of 4 and expect the indentation to be 4 spaces more than the try.
This should pass the indentation rule as it did in Checkstyle 8.40

### PR DESCRIPTION
**Pull request by Abhishek-kumar09**
Fixes #9341 
*This is an extension to the solution of https://github.com/checkstyle/checkstyle/issues/9236*

Diff Regression projects: https://gist.githubusercontent.com/Abhishek-kumar09/0745ecf11934e0214b9187f9d4896e4a/raw/29775c9e897e152f5cda2bdcc8cadfe57690d1a7/project-to-test-on.properties
Diff Regression config: https://gist.githubusercontent.com/Abhishek-kumar09/0745ecf11934e0214b9187f9d4896e4a/raw/29775c9e897e152f5cda2bdcc8cadfe57690d1a7/my_check.xml
Report label: Default Indentation profile with force Strict

### CODE REVIEW

**Comment by romani** [comment]
Please fix CI.
And please add user input as is, it is simple and very clear on error. Simple code is good to see how it should be.
Please run regression diff report

**Comment by Abhishek-kumar09** [comment]
Githu, generate report

**Comment by Abhishek-kumar09** [comment]
Github, generate report

**Comment by Abhishek-kumar09** [comment]
Github, generate report

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationTryBlock.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ */                                                                         //indent:1 exp:1
+
+package com.puppycrawl.tools.checkstyle.checks.indentation.indentation; //indent:0 exp:0
+
+import java.io.InputStreamReader; //indent:0 exp:0
+
+public class InputIndentationTryBlock { //indent:0 exp:0
+    public void delete(final int assetId) { //indent:4 exp:4
+        try { //indent:8 exp:8
+            new Object().equals(Integer.valueOf(assetId)); //indent:12 exp:12
```
Comment (romani) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationTryBlock.java:20]: please add one more sentence from 
https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/1526546_2021085309/reports/diff/Orekit/index.html#A6

to make sure it is covered.
We try to copy from regression reports all cases that are become different and it is not covered by our Inputs.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationTryBlock.java (hunk 1 above)
Comment (Abhishek-kumar09) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/indentation/indentation/InputIndentationTryBlock.java:20]: Done

**Review by romani — APPROVED** [review]

**Review by pbludov — APPROVED** [review]

**Comment by pbludov** [comment]
> https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/1526546_2021065525/reports/diff/index.html

Report is good. Guava violations are bogus due to two-space indentation.
Two new Orekit violations are excpected.

**Pull request merged**